clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/metrics/UrlMetricsInstrumentation.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.docker.registry.metrics
+
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import com.squareup.okhttp.Response
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import java.util.concurrent.TimeUnit
+
+import static java.lang.String.format
+
+@Slf4j
+@Component
+class UrlMetricsInstrumentation {
+
+  private final Registry registry
+
+  private final Id timingId
+  private final Id counterId
+
+  @Autowired
+  UrlMetricsInstrumentation(Registry registry) {
+    this.registry = registry
+    timingId = registry.createId('UrlRequestTime').withTag('className', UrlMetricsInstrumentation.simpleName)
+    counterId = registry.createId('UrlCount').withTag('className', UrlMetricsInstrumentation.simpleName)
+  }
+
+  void interceptorCompleted(Response response, long start, long end) {
+    registry.timer(timingId.withTag('url', response.request().url().toString())).record(end - start, TimeUnit.MILLISECONDS)
```

Comment:
As @mdirkse said: We should split `url` into `host` and `path`, and put them in separate tags. That way we can also do aggregations top level registry for instance.

Reply:
Updated. Metrics are now recorded with a `host` and `path` tag, instead of one `url` tag. Usability is limited in our own scenario, since `host` is always the same.